independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -72,6 +72,11 @@ public <T> T getIfActive(Contextual<T> contextual, Function<Contextual<T>, Creat
             // Bean instance does not exist - create one if we have CreationalContext
             instance = new ContextInstanceHandleImpl<T>((InjectableBean<T>) contextual,
                     contextual.create(creationalContext), creationalContext);
+            if (!instance.isAvailable()) {
```

Comment:
If I understand it correctly the _contextual instance_ (not `Bean`) may only be null if a `@RequestScoped` synthetic bean returns `null`. If so I think that it would make more sense to modify the `BeanGenerator.implementCreate()` method to check the non-null return value for normal-scoped synthetic beans. WDYT?

This would fix the problem with "uninitialized" `@ApplicationScoped` synthetic beans as well.

Reply:
You are the expert here 😉.

See the linked PR for the problem this PR attempts to address

Reply:
Ok, here's the pull request: https://github.com/quarkusio/quarkus/pull/29775 :-)